cognite_toolkit/cdf_tk/load.py

```diff
@@ -313,120 +197,260 @@ def load_transformations(
+        self, items: Group | Sequence[GroupLoader], ToolGlobals: CDFToolConfig, drop: bool
+    ) -> Group | GroupLoader | None:
+        created = self.client.iam.groups.create(items)
+        old_groups = self.client.iam.groups.list(all=True).data
+        created_names = {g.name for g in created}
+        to_delete = [g.id for g in old_groups if g.name in created_names]
+        self.client.iam.groups.delete(to_delete)
+        return created
+
+
+@final
+class DatapointsLoader(Loader[str, pd.DataFrame, list[pd.DataFrame]]):
+    load_files_individually = True
+    filetypes = frozenset({"csv", "parquet"})
+    api_name = "time_series.data"
+    folder_name = "timeseries_datapoints"
+    resource_cls = pd.DataFrame
+
+    @classmethod
+    def get_required_capability(cls, ToolGlobals: CDFToolConfig) -> Capability:
+        return TimeSeriesAcl(
+            [TimeSeriesAcl.Action.Read, TimeSeriesAcl.Action.Write],
+            TimeSeriesAcl.Scope.DataSet([ToolGlobals.data_set_id]),
+        )
+
+    @classmethod
+    def get_id(cls, item: T_Resource) -> T_ID:
+        raise NotImplementedError("Datapoints do not have an id")
+
+    def create(
+        self, items: pd.DataFrame | Sequence[pd.DataFrame], ToolGlobals: CDFToolConfig, drop: bool
+    ) -> pd.DataFrame | pd.DataFrame | None:
+        for item in items:
+            self.client.time_series.data.insert_dataframe(item)
+        return None
+
+    def load_file(self, filepath: Path, ToolGlobals: CDFToolConfig) -> T_Resource:
+        if filepath.suffix == ".csv":
+            return pd.read_csv(filepath, parse_dates=True, index_col=0)
+        elif filepath.suffix == ".parquet":
+            return pd.read_parquet(filepath, engine="pyarrow")
         else:
-            print(f"  Would have created {len(transformations)} transformation.")
-    except Exception as e:
-        print(f"[bold red]ERROR:[/] Failed to create transformations.\n{e}")
-        ToolGlobals.failed = True
+            raise ValueError(f"Not supported file type {filepath.suffix}")
+
+
+@final
+class RawLoader(Loader[str, pd.DataFrame, list[pd.DataFrame]]):
+    load_files_individually = True
+    filetypes = frozenset({"csv", "parquet"})
+    api_name = "raw.rows"
+    folder_name = "raw"
+    resource_cls = pd.DataFrame
+    actions = frozenset({})
+    acl = RawAcl
+    default_db: str = "default"
+
+    def __init__(self, client: CogniteClient):
+        super().__init__(client)
+        self.db = self.default_db
+        self.table = ""
+
+    @classmethod
+    def get_required_capability(cls, ToolGlobals: CDFToolConfig) -> Capability:
+        return RawAcl([RawAcl.Action.Read, RawAcl.Action.Write], RawAcl.Scope.All())
+
+    @classmethod
+    def get_id(cls, item: T_Resource) -> T_ID:
+        raise NotImplementedError("Raw data does not have an id")
+
+    def load_file(self, filepath: Path, ToolGlobals: CDFToolConfig) -> pd.DataFrame:
+        for pattern in [r"(\d+)\.(\w+)\.(\w+)", r"(\d+)\.(\w+)"]:
```

Comment:
Should we introduce explicit definition of raw tables through yaml files instead of reimplementing the old implicit based on csv data files? Declaring desired raw dbs and tables also simplifies error handling.

Reply:
I agree on this, same argument as I have for `space`s. But lets do that in another PR to keep this as pure refactoring.